exercises/05-reselect/src/modules/users/users-slice.ts

```diff
@@ -0,0 +1,33 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { UserData, User } from './user-types'
+
+export interface UsersState {
+  title: string
+  users: User[]
+}
+
+export const initialState: UsersState = {
+  title: 'User Management',
+  users: [],
+}
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState: initialState,
+  reducers: {
+    addUser: (state, { payload: user }: PayloadAction<UserData>) => ({
+      ...state,
+      users: [
+        ...state.users,
+        {
+          id: state.users.length + 1,
```

Comment:
I don't think there's any added value in storing the id here as it can easily be calcualated on the fly.

I know you are using it later as `key` for the list, but obviously the identification is natural sort of the array.

Reply:
And in all tutorials, it is said, that you should not use an array index as `key`. There are obviously exceptions, but in training, I would be explicit with id.

Reply:
I try to change the logic of the exercises as little as possible. 
I'd prefer to leave the `id` as it is.

I suppose the `id` is there to simulate an environment where the `id` would be something meaningful, but here in the training it's simplified.

Reply:
You both convinced me... good job :D 

Anyway 
>And in all tutorials, it is said, that you should not use an array index as key.

And I really don't like that, because people associate key with something that needs to be unique identifier and should never be natural array index, but that's not always the case, there are exceptions and people should be aware of them.